bitcoin/psbt.c

```diff
@@ -32,8 +32,12 @@ static struct wally_psbt *init_psbt(const tal_t *ctx, size_t num_inputs, size_t
 	else
 		wally_err = wally_psbt_init_alloc(0, num_inputs, num_outputs, 0, &psbt);
 	assert(wally_err == WALLY_OK);
+	tal_steal(ctx, psbt);
+	/* If both of these are zero, no sub-allocations were done */
+	if (num_inputs || num_outputs)
```

Comment:
Is there any harm in calling it unconditionally?

Reply:
No *harm*, but we have a debug check for the moment that we don't call tal_gather_wally unless there's something to gather.  Found some places where I was being overzealous.